constructor/fcp.py

```diff
@@ -109,6 +109,7 @@ def _fetch(download_dir, precs):
     if not isdir(download_dir):
         os.makedirs(download_dir)
     pc = PackageCacheData(download_dir)
+    pc.first_writable()
```

Comment:
`first_writable` is a class method so it might be misleading to use it as an instance method here. `download_dir` is not really used; it takes the value from `context.pkgs_dirs`, which has been incidentally set to `download_dir` 😬 

Anyway, this seems like the same solution I proposed in #548, right?

Reply:
yup, I agree with you. Let me test your PR.